plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/PlayAssetDeliveryTest.java

```diff
@@ -127,6 +129,109 @@ public void cancel_success() {
     assertThat(resultDict).isEqualTo(testDict);
   }
 
+  @Test
+  public void assetPackStateUpdatedTest() {
+    doAnswer(
+            invocation -> {
+              AssetPackStateUpdateListener listener =
+                  (AssetPackStateUpdateListener) invocation.getArguments()[0];
+              List<AssetPackState> packStateSequence =
+                  PlayAssetDeliveryTestHelper.createAssetPackStateList();
+              for (AssetPackState packState : packStateSequence) {
+                listener.onStateUpdate(packState);
+              }
+              return null;
+            })
+        .when(assetPackManagerMock)
+        .registerListener(any(AssetPackStateUpdateListener.class));
+
+    ArgumentCaptor<String> signalNameCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<Object> signalArgsCaptor = ArgumentCaptor.forClass(Object.class);
+
+    PlayAssetDelivery testSubject = spy(new PlayAssetDelivery(godotMock, assetPackManagerMock));
+
+    testSubject.registerAssetPackStateUpdatedListener();
+
+    verify(testSubject, times(3))
+        .emitSignalWrapper(signalNameCaptor.capture(), signalArgsCaptor.capture());
+
+    assertThat(signalNameCaptor.getAllValues())
+        .isEqualTo(
+            Arrays.asList(
+                "assetPackStateUpdated", "assetPackStateUpdated", "assetPackStateUpdated"));
+
+    List<Dictionary> expectedList =
+        PlayAssetDeliveryTestHelper.createAssetPackStateList()
+            .stream()
+            .map(
+                (AssetPackState state) ->
+                    PlayAssetDeliveryUtils.convertAssetPackStateToDictionary(state))
+            .collect(Collectors.toList());
+    assertThat(signalArgsCaptor.getAllValues()).isEqualTo(expectedList);
+  }
+
+  @Test
+  public void fetch_success() {
+    // Mock the side effects of Task<AssetPackStates> object, call onSuccessListener the instant
+    // it is registered.
+    Dictionary testDict = PlayAssetDeliveryTestHelper.createAssetPackStatesTestDictionary();
+    AssetPackStates testAssetPackStates = new AssetPackStatesFromDictionary(testDict);
+
+    Task<AssetPackStates> assetPackStatesSuccessTaskMock =
+        PlayAssetDeliveryTestHelper.createMockOnSuccessTask(testAssetPackStates);
+
+    PlayAssetDelivery testSubject = spy(new PlayAssetDelivery(godotMock, assetPackManagerMock));
+    when(assetPackManagerMock.fetch(anyListOf(String.class)))
+        .thenReturn(assetPackStatesSuccessTaskMock);
+
+    // Set up ArgumentCaptors to get the arguments received by emitSignalWrapper()
+    ArgumentCaptor<String> signalNameCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<Object> signalArgsCaptor = ArgumentCaptor.forClass(Object.class);
+
+    testSubject.fetch(new String[] {"pack1", "pack2"}, 16);
+
+    verify(testSubject).emitSignalWrapper(signalNameCaptor.capture(), signalArgsCaptor.capture());
+
+    assertThat(signalNameCaptor.getValue()).isEqualTo("fetchSuccess");
```

Comment:
Can you make PlayAssetDelivery.FETCH_SUCCESS package-private and reference it here? Same with fetchError in the other test.

Reply:
Used package-private const variables for all tests.